java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -220,19 +219,10 @@ private AccessToken refreshAccessToken(int attempt) throws IOException {
 
             CanonicalOAuthAccessTokenResponseDto tokenResponse;
 
-            Optional<AccessToken> sharedAccessToken = getSharedAccessTokenIfSupported();
-            if (this.currentToken == null) {
-                this.currentToken = sharedAccessToken.orElse(null);
-            }
+            AccessToken token = getSharedAccessTokenIfSupported().orElse(null);
 
-            if (sharedAccessToken.isPresent()) {
-                // we have a token, but shared token is newer. Other instance refreshed, so use it
-                if (sharedAccessToken.get().getExpirationTime().after(this.currentToken.getExpirationTime())) {
-                    this.currentToken = sharedAccessToken.get();
-                }
-            }
 
-            if (shouldRefresh(this.currentToken, clock.instant())) {
+            if (token == null || shouldRefresh(token, clock.instant())) {
```

Comment:
```suggestion
            if (shouldRefresh(token, clock.instant())) {
```
revert, `shouldRefresh` handles null case